src/main/java/com/minecolonies/colony/ColonyView.java

```diff
@@ -289,6 +298,30 @@ public IMessage handlePermissionsViewMessage(ByteBuf buf)
     }
 
     /**
+     * Update a ColonyView's workOrders given a network data ColonyView update packet.
+     * This uses a full-replacement - workOrders do not get updated and are instead overwritten.
+     *
+     * @param buf       Network data
+     * @param order     is 0 if this is the first message after a change.
+     * @return          null == no response
+     */
+    public IMessage handleColonyViewWorkOrderMessage(ByteBuf buf, int order, ColonyViewWorkOrderMessage.workOrderMessages operation)
+    {
+        if(order == 0)
```

Comment:
We shouldn't do this, we should edit each packet individually. I questioned what was happening with the i++ earlier, and now I think it should be removed. If you look below at how citizens are handled, that's a better way.

Also on the note of delete, we have a packet for remove citizen and remove building, so I think we should have one for remove work order because we are treating them similarly.